### DISCUSSION TITLE
Polling frequency for map update

### DISCUSSION THREAD

**Discussion by nhmariend** [Thomas55555/husqvarna_automower#299] [Ideas]
Is it possible to change the frequency ?
The mobile app seems to update every 5 seconds so its possible to see where the mower is located.

**Comment by Thomas55555** [comment]
How often do you get updates?

**Comment by Thomas55555** [comment]
But to answer you question: Normally, the position is updated when the mowers pushes a new location via websocket. So there is no polling. There is only a additional polling every 300s, to get the data at least, if the websocket is not working.

**Comment by nhmariend** [comment]
[image: image.png]
Hi here is how it look like now. Its charging and wait until tomorrow.

I added the picture with coordinates 2 hours ago or so.
Latitude

[image: image.png]

Longitude

[image: image.png]

Sometimes it takes 4-5 minus

On the phone it updates about every 5 seconds

I translated to Norwegian too yesterday. Did you see that ?
Almost the same as Danish

Best regards Niels Henrik Mariendal


tir. 12. jul. 2022 kl. 17:16 skrev Thomas55555 ***@***.***>:

**Comment by Thomas55555** [comment]
I can't see any pictures

**Reply by nhmariend** [reply]
I tried to ansver in mail, try on github now.
![latitude](https://user-images.githubusercontent.com/64189587/178542524-4cbd0023-60a0-4977-a201-a9a3c9775a14.png)
![longitude](https://user-images.githubusercontent.com/64189587/178542529-d9245c4d-e102-42bb-ac93-bbaba16100f9.png)
![Mower](https://user-images.githubusercontent.com/64189587/178542531-977a3a27-ff2b-4eb7-bfa0-8d3d5cea62fe.png)

**Comment by prairiesnpr** [comment]
I'm seeing position updates from the websocket right at 30 seconds when mowing. If you are seeing updates every 5 min, it makes me wonder if the websocket connection is active.

**Comment by nhmariend** [comment]
I will check my wireless network outside, maybe add an access point more

**Reply by Thomas55555** [reply]
> I will check my wireless network outside, maybe add an access point more

Aren't all the mowers sending their signal via cellular?

**Comment by prairiesnpr** [comment]
Interestingly, I was doing some dev work today and mine started doing the same thing, it was updating at a 5 min interval.  Reloading the integration resolved the issue, with home assistant actually slightly ahead of the app.

**Reply by Thomas55555** [reply]
> 
I'm also noticing, that the websocket reaction is quite poor since a few days. I also restarted HA, but it doesn't make it better. Sometimes it's start working again without a restart. But most of the updates I receive via REST

**Comment by nhmariend** [comment]
This is how it look now. 
![image](https://user-images.githubusercontent.com/64189587/179263359-a07bbba9-1752-4c22-a3b3-d8fce08c0286.png)


I tried Danish and swedish language in Home assistant and it work, but not norwegian. I still get English Mowing instead of Klipper.

It should be the same as Swedish

**Comment by Thomas55555** [comment]
I think, there really broke something, because there is a new message format now from the websocket. Shoud be fixed in `aioautomower==2022.7.2`
Testers welcome.

**Reply by prairiesnpr** [reply]
Moved to 2022.7.2, it's running 1 to 1 with the app so far, will keep an eye on it and see if it stays connected.

**Comment by nhmariend** [comment]
Updated to 2022.7.3 and run the mower one cycle.
Still slow to update.

I have 3 tp-link X60 mesh network. On in the basement and 2 in the main floor.
Its not a large house.
Could it be that I need better wireless outside?
It could be that some places is not so good, but my feeling is that its more frequent in tha spp on my iphone.
Sometimes I see it update every 1-2 seconds.

Or is it some place to see what happen inside HA ?`
Just hope to get the same as in the app.

I running on a 1 year old powerful NUC 

Home Assistant 2022.7.5
Supervisor 2022.07.0
Grensesnitt 20220707.1 – latest
 

![latitude2](https://user-images.githubusercontent.com/64189587/179831055-a9c07eab-3c5d-47bf-a36c-7a8c1b7af603.png)

**Comment by Thomas55555** [comment]
Looks normal. Your mower isn't connected to WiFi, isn't it?

Niels Henrik Mariendal ***@***.***> schrieb am Di., 19. Juli
2022, 21:34:

**Comment by prairiesnpr** [comment]
I think we have an issue here.

I added a sensor to track the update frequency and let the mower run last night.

![SmartSelect_20220720-062423_Home Assistant](https://user-images.githubusercontent.com/18637008/179960168-adf99b18-81ec-4b22-a139-29a273f5cf02.jpg)

As you can see, it mostly updated via the 300 second rest update.

Now more interestingly, mearly opening the Husqvarna mobile app and watching the map display will bring the web socket back.

![SmartSelect_20220720-062350_Home Assistant](https://user-images.githubusercontent.com/18637008/179960374-324862ba-c6a6-43b0-af85-7ba295e5682e.jpg)

That's what I did both times the updates resumed here, open the Husqvarna app and view the map and it resumed reporting.  Need to dig into it, but they must be doing something different to keep the update frequency. 

You can also reload the integration and the websocket updates will resume.

**Comment by nhmariend** [comment]
Sounds good that you look into it. I think you have made the same sensor as I have.

**Comment by prairiesnpr** [comment]
Finally heard back from Husqvarna.

------------------------------------------------------------------------------------------------

This is because there is a timeout between the mower and the cloud. To preserve data traffic the mower is not sending updates after some time and will only send updates every 15 minutes. Not sure if this can be changed when start a connection to Websockets. I will investigate this.

When using the Automower connect app we will override this save data. This will make the mower to use more mobile data traffic.

------------------------------------------------------------------------------------------------

Will see if they provide an option to do similar, but that explains the lack of updates.

**Reply by nhmariend** [reply]
Yes, could be nice if we can get the same in the app and in Home Assistant. Thanks for info

**Reply by Din-BH** [reply]
I guess no reply on this yet? Maybe good time to ask for an update... Would understand their reluctance to increase the data volumes given that they pay for the mobile network data - but on the other hand they are selling for a premium cost, so the service provided should be premium too!
